src/cxx/nux/xyz_to_mol.cpp

```diff
@@ -29,47 +28,93 @@ MODULE_CTOR(XYZToMolecule) {
 }
 
 MODULE_RUN(XYZToMolecule) {
-    const auto& [filename] = simde::MoleculeFromString::unwrap_inputs(inputs);
-
-    auto& z_from_sym  = submods.at("Z from symbol");
-    auto& atom_from_z = submods.at("Atom from z");
+    const auto& [xyz_data] = simde::MoleculeFromString::unwrap_inputs(inputs);
+    auto& z_from_sym       = submods.at("Z from symbol");
+    auto& atom_from_z      = submods.at("Atom from z");
 
+    std::string line;
     chemist::Molecule mol;
 
-    std::ifstream xyz_file(filename);
-
-    if(!xyz_file) { throw std::runtime_error("File not found: " + filename); }
+    std::stringstream buffer;
 
-    std::string line;
+    buffer << xyz_data;
+    if(buffer.str().empty()) {
+        throw std::runtime_error("File or XYZ data not valid, empty string");
+    }
 
-    // Skips the 1st line of the file, associated with the number of
-    // atoms in the file, and the 2nd line, the comment line.
-    std::getline(xyz_file, line);
-    std::getline(xyz_file, line);
+    int i = 0;
 
-    while(std::getline(xyz_file, line)) {
+    while(std::getline(buffer, line)) {
+        i++;
+        int num_atoms;
+        if(i == 1) {
+            try {
+                num_atoms = std::stoi(line);
+            } catch(const std::invalid_argument& e) {
+                throw std::runtime_error("Formatting of XYZ data incorrect");
+            }
```

Comment:
```suggestion
            num_atoms = std::stoi(line);
```

Only catch exceptions if you are going to handle them (trying to recover from the error).